packages/devtools_app/lib/src/screens/logging/logging_screen_v2/logging_table_v2.dart

```diff
@@ -0,0 +1,309 @@
+  }
+}
+
+class _LoggingTableRows extends StatefulWidget {
+  const _LoggingTableRows({
+    required this.model,
+  });
+
+  final LoggingTableModel model;
+
+  @override
+  State<_LoggingTableRows> createState() => _LoggingTableRowsState();
+}
+
+class _LoggingTableRowsState extends State<_LoggingTableRows> {
+  late final ScrollController _verticalController = ScrollController();
+
+  void onModelUpdate() {
+    setState(() {});
+  }
+
+  @override
+  void initState() {
+    widget.model.addListener(onModelUpdate);
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    widget.model.removeListener(onModelUpdate);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scrollbar(
+      thumbVisibility: true,
+      controller: _verticalController,
+      child: CustomScrollView(
+        controller: _verticalController,
+        slivers: <Widget>[
+          SliverVariedExtentList.builder(
+            itemCount: widget.model.filteredLogCount,
+            itemBuilder: (context, index) {
+              return LoggingTableRow(
+                index: index,
+                data: widget.model.filteredLogAt(index),
+                isSelected: false,
+              );
+            },
+            itemExtentBuilder: (index, _) =>
+                widget.model.getFilteredLogHeight(index),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+/// Shows and hides the context menu based on user gestures.
+///
+/// By default, shows the menu on right clicks and long presses.
+class _ContextMenuRegion extends StatefulWidget {
+  /// Creates an instance of [_ContextMenuRegion].
+  const _ContextMenuRegion({
+    required this.child,
+    required this.contextMenuBuilder,
+  });
+
+  /// Builds the context menu.
+  final ContextMenuBuilder contextMenuBuilder;
+
+  /// The child widget that will be listened to for gestures.
+  final Widget child;
+
+  @override
+  State<_ContextMenuRegion> createState() => _ContextMenuRegionState();
+}
+
+class _ContextMenuRegionState extends State<_ContextMenuRegion> {
+  Offset? _longPressOffset;
+
+  final ContextMenuController _contextMenuController = ContextMenuController();
+
+  static bool get _longPressEnabled {
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.android:
+      case TargetPlatform.iOS:
+        return true;
+      case TargetPlatform.macOS:
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.linux:
+      case TargetPlatform.windows:
+        return false;
+    }
+  }
+
+  void _onSecondaryTapUp(TapUpDetails details) {
+    _show(details.globalPosition);
+  }
+
+  void _onTap() {
+    if (!_contextMenuController.isShown) {
+      return;
+    }
+    _hide();
+  }
+
+  void _onLongPressStart(LongPressStartDetails details) {
+    _longPressOffset = details.globalPosition;
+  }
+
+  void _onLongPress() {
+    assert(_longPressOffset != null);
+    _show(_longPressOffset!);
+    _longPressOffset = null;
+  }
```

Comment:
long press is a mobile-focused gesture. Why are we using it here? We should be using right click or a three dot context menu since that is native for web.